src/pkg/utils/utils.go

```diff
@@ -708,23 +710,27 @@ func ResolveImageName(distro, image, release string) (string, string, error) {
 	logrus.Debugf("Release: '%s'", release)
 
 	if distro == "" {
-		distro = distroDefault
+		distro = viper.GetString("general.distro")
 	}
 
 	if distro != distroDefault && release == "" {
 		return "", "", fmt.Errorf("release not found for non-default distribution %s", distro)
 	}
 
 	if release == "" {
-		release = releaseDefault
+		release = viper.GetString("general.release")
 	}
 
 	if image == "" {
-		image = GetDefaultImageForDistro(distro, release)
+		if viper.IsSet("general.image") {
+			image = viper.GetString("general.image")
+		} else {
+			image = getDefaultImageForDistro(distro, release)
+		}
 	} else {
 		release = ImageReferenceGetTag(image)
 		if release == "" {
-			release = releaseDefault
+			release = viper.GetString("general.release")
```

Comment:
Once we potentially read `image` from the configuration file, the `release` needs to be extracted from it.

Earlier, `getDefaultImageForDistro` generated the `image` from the default `release`, but with a configuration file the release will be embedded in the `image`.

Reply:
Isn't this covered by `ImageReferenceGetTag()` called a bit earlier?

Reply:
Oops, now I see that function is not called in case the value is set in the config.

Reply:
Brain food: I wonder whether this exact line behaves correctly. If you run `podman pull registry.fedoraproject.org/fedora`, it'll try to pull `registry.fedoraproject.org/fedora:latest`. Note the `latest` tag. Maybe we should do the same instead of using the default.

Reply:
Yes, I thought about `latest` too. Not sure why we don't handle that. Let's ignore it, because it's an issue of its own.